extensions/pkg/webhook/certificates.go

```diff
@@ -108,23 +108,21 @@ func GenerateCertificates(ctx context.Context, mgr manager.Manager, certDir, nam
 }
 
 func generateNewCAAndServerCert(mode, namespace, name, url string) (*secrets.Certificate, *secrets.Certificate, error) {
-	caConfig := &secrets.CertificateSecretConfig{
+	caCert, err := (&secrets.CertificateSecretConfig{
 		CommonName: "webhook-ca",
 		CertType:   secrets.CACert,
-	}
-
-	caCert, err := caConfig.GenerateCertificate()
+	}).Generate()
```

Comment:
When calling `CertificateSecretConfig.Generate` you have to type cast to `*Certificate` every time now.
WDYT about keeping `CertificateSecretConfig.GenerateCertificate` as a shortcut for `Generate` + type cast?

Reply:
done with b7951b6cb